### ISSUE TITLE
sql: int4 type mapping error in npgsql

### ISSUE THREAD

**Issue by rkdrnfds** [cockroachdb/cockroach#24172]
When used cockroach db v2.0-beta.20180319,
it generates type mapping error as below when executed insertion query through Npgsql / EF Core

```
Npgsql.NpgsqlException (0x80004005): The PostgreSQL type 'int4', mapped to NpgsqlDbType 'Integer' isn't present in your database. You may need to install an extension or upgrade to a newer version.
```

cockroach db v2.0-alpha.20180129 does not generate those error and works well.

I created repository for repro so you can check it.
[repro github](https://github.com/rkdrnfds/array-example)

You can test it by manually creating database  
with table as below
```
CREATE TABLE "Chunks" (
	"Id" INT NOT NULL DEFAULT unique_rowid(),
	CONSTRAINT "PK_Chunks" PRIMARY KEY ("Id" ASC),
	FAMILY "primary" ("Id")
)
```
and runnning program.

**Comment by jordanlewis** [comment]
Hi @rkdrnfds,

I also don't understand this one. Like I said in #24173, we haven't changed our `pg_type` mapping recently so I don't expect this to have broken. We do provide the `int4` type correctly in `pg_type`. Thanks for the repro, I'll try to take a look at this later today.

**Comment by jordanlewis** [comment]
Hi @rkdrnfds,

I cannot reproduce this issue with your linked repository. As expected, the program successfully inserts the row into the database.

**Comment by bladefist** [comment]
@jordanlewis I reproduced this error, a couple times actually.  The error message happens when you issue a query against a table that you do not have permission to query.  It manifests itself as this error message, which sends the user off on the wrong path.

I've seen this happen a few times to me, and I always forget the solution and have to figure it out again.   To my future self who will google this again some day, hope things are well! :)

**Comment by jordanlewis** [comment]
This has been solved in 19.1.1. #24173.
